File: java/Solutions/longestCommonPrefixClass.java
package Solutions;
import java.util.Arrays;

public class longestCommonPrefixClass {
    public longestCommonPrefixClass() {
        super();
    }

    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs, 0, strs.length, (String a, String b) ->Integer.compare(a.length(), b.length()));
        String best = "";

        for (int i = 1; i <= strs[0].length(); i++)
        {
            String sub = strs[0].substring(0, i);
            Boolean contains = true;
            for (int j = 0; j < strs.length; j++){
                if(!strs[j].startsWith(sub)){
                    contains = false;
                    break;
                }
            }
            
            if(contains == false) break;
            best = sub;
        }

        return best;
    }
}

File: C#/Solutions/LongestCommonPrefixClass.cs
class LongestCommonPrefixClass
{
    public string LongestCommonPrefix(string[] strs)
    {
        var smalestWord = strs.OrderBy(x => x.Length).First();
        string best = "";

        for (int i = 1; i < smalestWord.Length; i++)
        {
            var sub = smalestWord.Substring(0, i+1);
            if (strs.Select(x => x.StartsWith(sub)).Contains(false)) break;
            best = sub;
        }

        return best;
    }
}